### DISCUSSION TITLE
What's the best strategy to migrate from fumadocs API 4.x.x to 5.x.x?

### DISCUSSION THREAD

**Discussion by chris4490** [fuma-nama/fumadocs#898] [General]
We started with fumadocs open API 4.x.x, but now we want to move to 5.x.x. We're hitting issues with imports.

```
./base/components/elements/api-playground/api-playground.helpers.ts:2:15
Type error: Module '"fumadocs-openapi"' declares 'PrimitiveRequestField' locally, but it is not exported.

  1 | /* eslint-disable security/detect-object-injection */
> 2 | import type { PrimitiveRequestField, ReferenceSchema, RequestSchema } from 'fumadocs-openapi';
```

I see that these items are [internally exported](https://github.com/search?q=repo%3Afuma-nama%2Ffumadocs%20PrimitiveRequestField&type=code) within the fumadocs-openapi, but not sure how to adjust our code to work with the new version.

**Comment by fuma-nama** [comment]
Hmm I'm not quite sure what your `api-playground.helpers.ts` does, mind sharing it?

**Comment by chris4490** [comment]
Much appreciated in the quick reply.

[api-playground.zip](https://github.com/user-attachments/files/17151856/api-playground.zip)


There's other import issues as well, like the mdx-components.tsx also has 
```
import {
  Root,
  API,
  APIInfo,
  APIExample,
  Responses,
  Response,
  ResponseTypes,
  ExampleResponse,
  TypeScriptResponse,
  Property,
  ObjectCollapsible,
  Requests,
  Request,
} from 'fumadocs-openapi/ui';
```

**Reply by fuma-nama** [reply]
the components are no longer recommended for use, they are supposed to be generated with Fumadocs OpenAPI's renderer.
Instead of using them, you should use `APIPage` directly.

You may also clone the source code if you want the components but don't need the automation features

**Comment by olabrahmi** [comment]
@fuma-nama any update on this? This is quite a big blocker for us...
